SabaCenter/ViewModel/PageViewModels/SchedulePageViewModel.swift

```diff
@@ -18,17 +17,18 @@ class SchedulePageViewModel: ViewModelBase {
         self.scheduleRepo = scheduleRepo
     }
 
-    func getListOfScheduleItems() -> Observable<[ScheduleViewModel]> {
+    func getListOfScheduleItems() -> [ScheduleViewModel] {
 
-        return self.scheduleRepo.getScheduleItems().map { schedules in
-            return schedules.map { item in
-                return ScheduleViewModel(schedule: item)
-            }
+        var scheduleItems = [ScheduleViewModel]()
+
+        for scheduleItem in self.scheduleRepo.getScheduleItems() {
```

Comment:
use `.map` instead. i.e. `self.scheduleRepo.getScheduleItems().map { item in ScheduleViewModel(schedule: item) }`

Reply:
Same for the other PageViewModels (LecturePageViewModel, ...)